src/VisualStudio/Xaml/Impl/Implementation/XamlProject.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.Xaml;
+using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem;
+using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.Legacy;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.LanguageServices.Xaml
+{
+    [DebuggerDisplay("{DisplayName,nq}|{nameof(XamlProject),nq}, {Id}")]
```

Comment:
It's generally frowned upon to have anything more than a single property/method call here for two reasons:

1. More Func-evals means worse stepping perf.  Ask me about the time I change the `DebuggerDisplay` for `System.Drawing.Size` someday...
2. Different languages (EEs) might interpret the strings differently.

We generally have a private method* at the end of the class called `GetDebuggerDisplay()`, and then have the attribute just be `[{GetDebuggerDisplay(),nq}]`.

*So that it doesn't get redundantly evaluated when the type is expanded in the watch window.